util/billing.go

```diff
@@ -0,0 +1,154 @@
+package util
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/olivere/elastic/v7"
+)
+
+var TierValidity int64
+
+const (
+	envEsURL       = "ES_CLUSTER_URL"
+	arcIdentifier  = "ARC_ID"
+	emailID        = "EMAIL"
+	subscriptionID = "SUBSCRIPTION_ID"
+)
+
+// Middleware function, which will be called for each request
+func BillingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		currentTime := time.Now().Unix()
```

Comment:
We shouldn't rely on system time from client and instead return an absolute remaining time back from the API response.